### PR TITLE
Limit Memory Usage When Logging Req/Resp Bodies

### PR DESCRIPTION
**Pull request by andrewhoff**
Swapped the `ioutil.ReadAll()` call here for a `LimitReader` to cap memory usage per logged body.
Also added a bit of extra test coverage to ensure that `logBody` is resetting the Reader after logging it, as well as added coverage with a large request/response body.

### CODE REVIEW

**Comment by andrewhoff** [comment]
Hey @gordcurrie I made this for you. 👋

File: goshopify.go [hunk 1]
```diff
@@ -32,6 +32,9 @@ const (
 var (
 	// version regex match
 	apiVersionRegex = regexp.MustCompile(`^[0-9]{4}-[0-9]{2}$`)
+
+	// MaxLoggedHTTPBodyBytes - limits the memory usage per logged request or response body to 32KiB
+	MaxLoggedHTTPBodyBytes int64 = 32000 // ~32KiB default to something more than reasonable, but expose it for configuration
```
Comment (luthermonson) [inline goshopify.go:37]: package var works but could be more useful to config per client. i needed flexibility for multi clients and updated a bunch of things to accommodate that so I'd like to at least advocate for that concept. perhaps adding another option to set this like WithLogger and storing it on the client struct?

**Comment by andrewhoff** [comment]
Ah that's a good point, I take a stab at that tomorrow evening, as I'm on
PDT. Thanks for the feedback!

**Comment by gordcurrie** [comment]
Thanks @andrewhoff , it's a bit crazy here I'll have a look at getting this in over the weekend, really appreciate you taking the time to get this in. 👋

**Comment by andrewhoff** [comment]
For sure! I will take a look at incorporating @luthermonson 's suggestion
in the meantime.

**Comment by gordcurrie** [comment]
Looks like `strings.Builder` was added in `1.10` I think it's reasonable to require `1.10`+ but I'll need to run that by some people before updating.

**Pull request closed without merging**